Add platform developer section to tasks index (#1076)
kubernetes-sigs/kueue: site/content/en/docs/tasks/_index.md

A _batch administrator_ manages the cluster infrastructure and establishes
quotas and queues.

As a batch administrator, you can learn how to:

- [Setup role-based access control](/docs/tasks/rbac)
  to Kueue objects.
- [Administer cluster quotas](/docs/tasks/administer_cluster_quotas) with ClusterQueues and LocalQueues.
- Setup [Sequential Admission with Ready Pods](/docs/tasks/setup_sequential_admission).

### Batch user

A _batch user_ runs [workloads](/docs/concepts/workload). A typical
batch user is a researcher, AI/ML engineer, data scientist, among others.

As a batch user, you can learn how to:
- [Run a Kueue managed batch/Job](/docs/tasks/run_jobs).
- [Run a Kueue managed Flux MiniCluster](/docs/tasks/run_flux_minicluster).
- [Run a Kueue managed Kubeflow MPIJob](/docs/tasks/run_mpi_jobs).
- [Run a Kueue managed KubeRay RayJob](/docs/tasks/run_rayjobs).
- [Submit Kueue jobs from Python](/docs/tasks/run_python_jobs).

### Platform developer

> A _Platform developer_ integrates Kueue with other software and/or contributes to Kueue.

Reviewer: ```suggestion
A _platform developer_ integrates Kueue with other software and/or contributes to Kueue.
```

- As a batch user, you can learn how to [run a Kueue managed Kubeflow MPIJob](/docs/tasks/run_mpi_jobs).
- As a batch user, you can learn how to [run a Kueue managed KubeRay RayJob](/docs/tasks/run_rayjobs).
- As a batch developer user, you can learn how to [submit Kueue jobs from Python](/docs/tasks/run_python_jobs).

### Platform Developer

> A _Platform Developer_ contributes to Kueue, including code, documentation and etc..

Reviewer: ```suggestion
A _Platform Developer_ integrates Kueue with other software and/or contributes to Kueue.
```
